edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/ExperimentalFeatures.cs

```diff
@@ -28,6 +28,16 @@ public static ExperimentalFeatures Create(IConfiguration experimentalFeaturesCon
             return experimentalFeatures;
         }
 
+        public static bool IsViaBrokerUpstream(ExperimentalFeatures experimentalFeatures, bool hasGatewayHostname)
```

Comment:
Nit: this doesn't belong in ExperimentalFeatures. Pls move it back to DependencyManager.

Reply:
it was moved here because there was a code duplication, so somewhere a common class is needed to be shared (this is called from dependency manager and from program.cs. In program.cs it is needed to decide if configUpdater should be awaited or not - if the communication is ViaBroker, then it should no wait.
Although the duplication is not big (so would not be a problem to copy), because it depends on config settings, odds are big that with the change of the config (e.g. it is not experimental feature anymore), one place will be fixed but not the other, seemed safer to merge. And because almost all components of that condition touches experimentalFetaures. and this was a static class already, it seemed a good candidate to move the condition here.
I think it makes sense the question that "is mqtt broker enabled' is very similar to the question that 'is the upstream communication goes through the new broker"